File: appli-android/app/src/main/java/com/example/sportapp/core/data/model/Equipment.kt
package com.example.sportapp.core.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.sportapp.core.utils.CustomDateUtils.getNowISO8601
import java.util.UUID

@Entity(
    tableName = "equipments",
    indices = [androidx.room.Index(value = ["uuid"], unique = true)]
)
data class Equipment(
    @PrimaryKey
    @ColumnInfo(name = "uuid") val uuid: String,

    val name: String,

    @ColumnInfo(defaultValue = "0") val synced: Boolean = false,
    @ColumnInfo(defaultValue = "0") val pendingDeletion: Boolean = false,

    @ColumnInfo(name = "updated_at") val updatedAt: String? = getNowISO8601(),
)

File: serveur/app/models/equipment.py
from sqlalchemy import Column, Integer, String, DateTime, func
from app.database import Base

import uuid

class Equipment(Base):
    __tablename__ = "equipments"

    id = Column(Integer, primary_key=True, index=True)
    uuid = Column(String, nullable=False, default=lambda: str(uuid.uuid4()), unique=True)
    
    name = Column(String, unique=True, nullable=False)

    updated_at = Column(DateTime(timezone=True), nullable=True)
